core/trino-main/src/main/java/io/trino/connector/WorkerDynamicCatalogManager.java

```diff
@@ -137,19 +147,26 @@ public void pruneCatalogs(Set<CatalogHandle> catalogsInUse)
             catalogCleanUpLock.unlock();
         }
 
-        // todo do this in a background thread
-        for (CatalogConnector removedCatalog : removedCatalogs) {
-            try {
-                removedCatalog.shutdown();
-            }
-            catch (Throwable e) {
-                log.error(e, "Error shutting down catalog: %s".formatted(removedCatalog));
+        var catalogsShutDown = Futures.whenAllComplete(removedCatalogs.stream()
+                .map(removedCatalog -> Futures.submit(
+                        () -> {
+                            try {
+                                removedCatalog.shutdown();
+                            }
+                            catch (Throwable e) {
+                                log.error(e, "Error shutting down catalog: %s".formatted(removedCatalog));
+                            }
+                        },
+                        executor))
+                .collect(toImmutableList()));
+
+        var ignored = catalogsShutDown.run(() ->
+        {
+            if (!removedCatalogs.isEmpty()) {
+                List<String> sortedHandles = removedCatalogs.stream().map(connector -> connector.getCatalogHandle().toString()).sorted().toList();
+                log.debug("Pruned catalogs: %s", sortedHandles);
```

Comment:
I'd just log the debug message for each individual catalog, just like we log each individual add above. Then this entire section can be removed.  Also I simplify the logging so you only get the "error pruning catalog" or the debug "pruned catalog" message.